tests/tests_geomstats/test_hypersphere.py

```diff
@@ -1,911 +1,701 @@
-
-    def test_squared_norm_and_squared_dist(self):
-        """
-        Test that the squared distance between two points is
-        the squared norm of their logarithm.
-        """
-        point_a = 1.0 / gs.sqrt(129.0) * gs.array([10.0, -2.0, -5.0, 0.0, 0.0])
-        point_b = 1.0 / gs.sqrt(435.0) * gs.array([1.0, -20.0, -5.0, 0.0, 3.0])
-        log = self.metric.log(point=point_a, base_point=point_b)
-        result = self.metric.squared_norm(vector=log)
-        expected = self.metric.squared_dist(point_a, point_b)
-
+    def test_extrinsic_to_angle(self, dim, point, expected):
+        space = self.space(dim)
+        result = space.extrinsic_to_angle(point)
         self.assertAllClose(result, expected)
 
-    def test_squared_dist_vectorization_single_sample(self):
-        one_point_a = self.space.random_uniform()
-        one_point_b = self.space.random_uniform()
-
-        result = self.metric.squared_dist(one_point_a, one_point_b)
-        self.assertAllClose(gs.shape(result), ())
-
-        one_point_a = gs.to_ndarray(one_point_a, to_ndim=2)
-        result = self.metric.squared_dist(one_point_a, one_point_b)
-        self.assertAllClose(gs.shape(result), (1,))
-
-        one_point_b = gs.to_ndarray(one_point_b, to_ndim=2)
-        result = self.metric.squared_dist(one_point_a, one_point_b)
-        self.assertAllClose(gs.shape(result), (1,))
-
-        one_point_a = self.space.random_uniform()
-        result = self.metric.squared_dist(one_point_a, one_point_b)
-        self.assertAllClose(gs.shape(result), (1,))
-
-    def test_squared_dist_vectorization_n_samples(self):
-        n_samples = self.n_samples
-
-        one_point_a = self.space.random_uniform()
-        one_point_b = self.space.random_uniform()
-        n_points_a = self.space.random_uniform(n_samples=n_samples)
-        n_points_b = self.space.random_uniform(n_samples=n_samples)
-
-        result = self.metric.squared_dist(one_point_a, one_point_b)
-        self.assertAllClose(gs.shape(result), ())
-
-        result = self.metric.squared_dist(n_points_a, one_point_b)
-        self.assertAllClose(gs.shape(result), (n_samples,))
-
-        result = self.metric.squared_dist(one_point_a, n_points_b)
-        self.assertAllClose(gs.shape(result), (n_samples,))
-
-        result = self.metric.squared_dist(n_points_a, n_points_b)
-        self.assertAllClose(gs.shape(result), (n_samples,))
-
-        one_point_a = gs.to_ndarray(one_point_a, to_ndim=2)
-        one_point_b = gs.to_ndarray(one_point_b, to_ndim=2)
-
-        result = self.metric.squared_dist(n_points_a, one_point_b)
-        self.assertAllClose(gs.shape(result), (n_samples,))
-
-        result = self.metric.squared_dist(one_point_a, n_points_b)
-        self.assertAllClose(gs.shape(result), (n_samples,))
-
-        result = self.metric.squared_dist(n_points_a, n_points_b)
-        self.assertAllClose(gs.shape(result), (n_samples,))
-
-    def test_norm_and_dist(self):
```

Comment:
missing? this could become a test in the RiemannianMetricParametrizer.

Reply:
(actually, this one seems useless since there is test_squared_dist_and_squared_norm already)

Reply:
already part of improvos